sklearn/decomposition/_nmf.py

```diff
@@ -823,21 +845,35 @@ def _fit_multiplicative_update(
     for n_iter in range(1, max_iter + 1):
         # update W
         # H_sum, HHt and XHt are saved and reused if not update_H
-        delta_W, H_sum, HHt, XHt = _multiplicative_update_w(
-            X, W, H, beta_loss, l1_reg_W, l2_reg_W, gamma, H_sum, HHt, XHt, update_H
+        W, H_sum, HHt, XHt = _multiplicative_update_w(
+            X,
+            W,
+            H,
+            beta_loss=beta_loss,
+            l1_reg_W=l1_reg_W,
+            l2_reg_W=l2_reg_W,
+            gamma=gamma,
+            H_sum=H_sum,
+            HHt=HHt,
+            XHt=XHt,
+            update_H=update_H,
         )
-        W *= delta_W
 
         # necessary for stability with beta_loss < 1
         if beta_loss < 1:
             W[W < np.finfo(np.float64).eps] = 0.0
 
         # update H
```

Comment:
These comments where there prior to this PR. I actually find them somewhat useful because it allows to visualize the structure of the algorithm at first sight. I added the info that it's only at fit or fit_transform to make it appear more useful :)